[WB-1849] Rework action semantic color tokens

jandrade opened the pull request:
## Summary:

Reworked `semanticColor` actions to use a new structure. Every `action` now
includes `default`, `hover` and `press` states, and each state includes
`border`, `background` and `foreground` tokens.

By doing this change, we'll offer a more consistent and predictable styling
direction for our interactive components.

Issue: https://khanacademy.atlassian.net/browse/WB-1849

## Test plan:

Navigate to `/?path=/docs/foundations-using-color--docs` and check that the
`Action` section includes all the new tokens and the structure is readable and
makes sense.

beaesguerra commented on packages/wonder-blocks-tokens/src/tokens/semantic-color.ts:
```diff
@@ -1,18 +1,87 @@
+                border: color.activeBlue,
+                background: color.activeBlue,
+                foreground: color.white,
+            },
```
Should the `focus` state also have tokens associated with it?

jandrade replied:
`focus` will have its own separate state, that is directly allocated in the `semanticColor.border.focus` token. This is something Caitlyn and I have discussed and the goal is that any focus on any component should use a blue outline, so that's the reason why I'm not including it here.

beaesguerra replied:
Sounds good! There is also the disabled focus outline!

beaesguerra commented on packages/wonder-blocks-tokens/src/tokens/semantic-color.ts:
```diff
@@ -1,18 +1,87 @@
+                foreground: color.activeRed,
+            },
         },
         disabled: {
```
Should the tokens associated with the `disabled` state also have the backgorund, foreground, border structure?

jandrade replied:
It might be in the future, but for now I'm leaving them out of every specific semantic global token and I'm defining them inside `Button` for now. Buttons need simplification and before doing that, I prefer to keep it this way. FWIW Design also has this structure, but this could change in the next weeks.

beaesguerra commented on packages/wonder-blocks-tokens/src/tokens/semantic-color.ts:
```diff
@@ -1,18 +1,87 @@
-            default: color.blue,
-            active: color.activeBlue,
+        // For primary actions
+        progressive: {
```
For my own understanding, what was the reason for using the `progressive`/`progressiveInverse` terminology instead of `primary`/`secondary`? 

And should we start using this terminology for component `kind` variants too?

beaesguerra reviewed (approved):
Looks good! Had some non-blocking questions I wanted to get your thoughts on! 

I noticed in the docs some of the token names wrap (you mentioned this also!). I confirmed that copy + pasting it still copies the name without the line break though!
<img width="1009" alt="image" src="https://github.com/user-attachments/assets/64ac0f33-410d-4014-9a85-fd229fe5cb66" />

jandrade commented:
@beaesguerra I made some changes based on your feedback/questions (also after discussing offline).

- Changed the structure to use `filled` and `outlined` to represent `primary/semantic` values (this might change in the future depending on the visuals approved for polaris).
- Changed the structure of the "Using color" page to include an example of how all these new tokens work in their context + modified the layout to make it more readable.

https://5e1bf4b385e3fb0020b7073c-xynstbxfft.chromatic.com/?path=/docs/foundations-using-color--docs

<img width="1092" alt="Screenshot 2025-01-21 at 5 17 51 PM" src="https://github.com/user-attachments/assets/e6309f55-0dcd-45a3-8298-7c1339388297" />

Let me know what you think, and thanks again for the suggestions!

beaesguerra reviewed (approved):
Looks great! Thanks for making those updates! 🎉
